File: generate_ppt.R
library(dplyr)
library(reticulate)

# source edit_pres function
reticulate::source_python("edit_pres.py")

# load in data files
for (file in list.files("data")) {
  splits <- strsplit(file, "\\.")
  assign(splits[[1]][1],
         read.csv(paste0("data/", file)))
} 

# rowwise mutate a list column onto chart_df containing the table data for each group
full_data <- chart_df %>% 
  rowwise() %>% 
  dplyr::mutate(table = list(get(paste0("table_", group))))

# create outputs directory if it doesn't exist
if (!dir.exists("outputs")) {
  dir.create("outputs")
}

# rowwise mutate the edit_pres function to generate parametrized powerpoint
generate_ppt <- full_data %>% 
  rowwise() %>% 
  dplyr::mutate(
    ppt = edit_pres(group, ., table, "templates/ppt-template.pptx", 
                                paste0("outputs/report_group_", group, ".pptx"))
    )

File: generate_ppt.py
import pandas as pd
import glob
import os

# source script with edit_pres function
exec(open('edit_pres.py').read())

# load in data files
path = "data/"
data_files = glob.glob(os.path.join(path, "*.csv"))
dataframes = dict()
for i in data_files:
    path = os.path.basename(i)
    name = path.split('.')[0]
    dataframes[name] = pd.read_csv("data/" + name + ".csv")

# create outputs directory if it doesn't exist
if not os.path.isdir("outputs"):
  os.mkdir("outputs")

# generate a ppt for each group
for group in dataframes['chart_df'].group.values:
  tables_data = dataframes['table_' + group]
  edit_pres(group, dataframes['chart_df'], tables_data, 
  'templates/ppt-template.pptx', 'outputs/results_group_' + group + '.pptx')
  
